app/main/routes_frontend.py

```diff
@@ -270,7 +270,11 @@ def _pico_recipes():
     global pico_recipes
     pico_recipes = load_pico_recipes()
     recipes_dict = [json.loads(json.dumps(recipe, default=lambda r: r.__dict__)) for recipe in pico_recipes]
-    return render_template('pico_recipes.html', recipes=recipes_dict, invalid=invalid_recipes.get(MachineType.PICOBREW, set()))
+    if display_temperature() == 'C':
+        for recipe in recipes_dict:
+            for step in recipe['steps']:
+                step['temperature'] = calcDisplayTemp(step['temperature'])              
+    return render_template('pico_recipes.html', recipes=recipes_dict, invalid=invalid_recipes.get(MachineType.PICOBREW, set()), display_temp=display_temperature())

```

Comment:
I'd recommend for simplicity to just handle all F->C conversions client side vs having to wire everything into the python code like this.

If we did that there wouldn't be any need to "test with C/S/Pro/Zy/Z" instead it would be create recipe using your units of choice, they get stored in the server with F as the units. Later if we wanted to add the ability to import recipes using C vs F that can be handled at a layer of indirection via an import experience translating the BeerXML temperature units into the desired storage format.